contents/testing/integration-testing.md

```diff
@@ -0,0 +1,212 @@
+Integration tests find these discrepancies before any serious damage is done.
+</box>
+
+Although system testing also help to uncover faults between modules, integration testing is still highly encouraged as it is easier to test, isolate failures and run faster than a system test.
+More advantages of integration testing before system testing can be found [here](https://bitwaretechnologies.com/integration-testing-important-software-testing-life-cycle/).
+
+Hence, integration testing is an important process which should be applied in the development of most softwares products such as web, mobile and desktop applications. 
+Concepts shared in this article can be adopted by other different software applications.
+
+On top of that, in the current era of <tooltip content="Practice of merging all developers work  into a shared repository">Continuous Integration</tooltip> (CI), integration testing is useful to help early detecion of defects while maintaining the quality of the product throughout further developments.
+The concept of [Continuous Integration Testing](https://crossbrowsertesting.com/blog/continuous-integration/continuous-integration-testing-delivery/) is also formed to combine integration testing with continuous integration.
+
+:fas-info-circle: More information on instances where integration testing is important can be found on [this article](https://www.guru99.com/integration-testing.html).
+
+## How does integration testing work?
+
+Using this specific system as an example, below are a list of things you should follow to execute effective integration testing:
+
+1. Prepare the integration test plan.
+1. Design the test scenarios, cases and scripts.
+1. Executing the test scripts and report defects if any.
+1. Tracking defects and re-testing application.
+
+### 1. Prepare the integration test plan
+
+Before the actual implementation of your integration tests, it is important to decide on the approach used. There are a few different approaches of integration testing in which you can adopt depending on the development progress of your application.
+
+#### [Big bang approach](https://www.tutorialspoint.com/software_testing_dictionary/big_bang_testing.htm):
+
+This approach involves integrating all the components in your design diagram together and testing everything at once in a complete state. 
+This is convenient but it is difficult to isolate defects and there is a high chance of missing critical underlying defects.
+Big bang integration testing is usually used for smaller applications with few components.
+
+An example of how big bang integration testing can be applied to our given example is shown below:
+
+<center>
+
+![Big bang integration testing diagram](integration-test/big-bang-integration.png "Big bang integration testing diagram]")
+</center>
```

Comment:
the font size in the image should follow the main article. In this case, it has a caption that appears bigger than the article headings.